common/tools/dev-tool/src/commands/testNode.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { leafCommand, makeCommandInfo } from "../framework/command";
+import concurrently from "concurrently";
+import { config } from "dotenv";
+import { isProxyToolActive } from "../util/testProxyUtils";
+config();
+
+export const commandInfo = makeCommandInfo(
+  "test:node",
+  "runs the node tests using mocha with the default and the provided options; starts the proxy-tool in record and playback modes",
+  {
+    "mocha-options": {
+      kind: "string",
+      description: "Mocha options along with the test files(glob pattern) as expected by mocha",
+      default: ""
+    }
+  }
+);
+
+export default leafCommand(commandInfo, async (_) => {
+  if (process.argv[3] !== "--mocha-options" && !process.argv[4]) {
+    throw new Error(
+      "unexpected command provided; expected = `dev-tool test:node --mocha-options '<options>'`"
+    );
+  }
+
+  const testProxyStart = "dev-tool test-proxy start";
+  const mochaCMDWithDefaults =
+    "nyc mocha -r esm --require ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --full-trace";
+  const mochaCommand = `${mochaCMDWithDefaults} ${process.argv[4]}`;
+
+  let runOnlyMochaCommand = false; // Boolean to figure out if we need to run just the mocha command or the test-proxy too
+
+  const mode = process.env.TEST_MODE;
+  if (mode === "live") {
+    runOnlyMochaCommand = true; // No need to start the proxy tool in the live mode
+  } else {
+    try {
+      await isProxyToolActive();
+      // No need to run a new one if it is already active
+      // Especially, CI uses this path
+      console.log(
```

Comment:
moved to log.info to go with the labels from dev-tool since I'll be moving away from concurrently in the future PRs to be able to start the processes programmatically instead of the commands.